core/scale/collection.hpp

```diff
@@ -75,10 +75,13 @@ namespace kagome::scale::collection {
    * same specified type
    * @tparam T collection item type
    * @param stream source stream
+   * @param decode a functor that decodes an object of type T
    * @return decoded collection or error
    */
   template <class T>
-  outcome::result<std::vector<T>> decodeCollection(common::ByteStream &stream) {
+  outcome::result<std::vector<T>> decodeCollection(
+      common::ByteStream &stream,
+      std::function<outcome::result<T>(common::ByteStream &)> decode_f) {
```

Comment:
why did you change scale encoding in this PR?

Reply:
Oops, accidental clang-format. I will fix it.